source/core/mesh_controller.cc

```diff
@@ -0,0 +1,51 @@
+//
+// Created by luckab on 12/07/23.
+//
+
+#include <core/mesh_controller.h>
+
+
+using namespace std;
+double
+MeshController::calculate_coarsening_factor(int current_number_of_cell)
+{
+  // Parameter of the controller
+  double P = 0.5;
+  double I = 0.3;
+  double D = 0.1;
```

Comment:
Will these parameters always work (i.e., could it be interesting to add them in the .prm)?